pack.go

```diff
@@ -23,56 +23,108 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/opencontainers/go-digest"
 	specs "github.com/opencontainers/image-spec/specs-go"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
-	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
 	"oras.land/oras-go/v2/content"
 	"oras.land/oras-go/v2/errdef"
 )
 
-// MediaTypeUnknownConfig is the default mediaType used when no
-// config media type is specified.
-const MediaTypeUnknownConfig = "application/vnd.unknown.config.v1+json"
+const (
+	// MediaTypeUnknownConfig is the default mediaType used when no
+	// config media type is specified.
+	MediaTypeUnknownConfig = "application/vnd.unknown.config.v1+json"
+	// MediaTypeUnknownArtifact is the default artifactType used when no
+	// artifact type is specified.
+	MediaTypeUnknownArtifact = "application/vnd.unknown.artifact.v1"
+)
 
 var (
-	// ErrMissingArtifactType is returned by PackArtifact() when no artifact
-	// type is specified.
-	ErrMissingArtifactType = errors.New("missing artifact type")
-	// ErrInvalidDateTimeFormat is returned by PackArtifact() when
-	// AnnotationArtifactCreated is provided, but its value is not in RFC 3339
-	// format.
-	// Reference: https://datatracker.ietf.org/doc/html/rfc3339#section-5.6
+	// ErrInvalidDateTimeFormat is returned by Pack() when
+	// AnnotationArtifactCreated or AnnotationCreated is provided, but its value
+	// is not in RFC 3339 format.
+	// Reference: https://www.rfc-editor.org/rfc/rfc3339#section-5.6
 	ErrInvalidDateTimeFormat = errors.New("invalid date and time format")
 )
 
 // PackOptions contains parameters for oras.Pack.
 type PackOptions struct {
+	// Subject is the subject of the manifest.
+	Subject *ocispec.Descriptor
+	// ManifestAnnotations is the annotation map of the manifest.
+	ManifestAnnotations map[string]string
+
+	// PackImageManifest controls whether to pack an image manifest or not.
+	//   - If true, pack an image manifest; artifactType will be used as the
+	// the config descriptor mediaType of the image manifest.
+	// media type.
+	//   - If false, pack an artifact manifest.
+	// Default: false.
+	PackImageManifest bool
 	// ConfigDescriptor is a pointer to the descriptor of the config blob.
+	// If specified, artifactType will be implied by the media type of the
+	// specified descriptor, and ConfigAnnotations will be ignored.
+	// This option is valid only when PackImageManifest is true.
```

Comment:
Can we use `ConfigDescriptor==nil` to imply`PackImageManifest==true`?

Reply:
Please resolve this, in the fallback case, we might need to use `PackImageManifest=true` and `ConfigDescriptor!=nil` to store artifact type